### ISSUE TITLE
Crash on OS X using floating texture window layouts

### ISSUE THREAD

**Issue by jdolan** [TTimo/GtkRadiant#335]
Any of the GtkRadiant layouts which use a floating texture browser window will crash after showing the texture window a few times. The problem appears to be upstream, in that GtkGLExt and XQuartz do not agree when swapping GL contexts / making one current:

I/O warning : failed to load external entity "/Users/ju/GtkRadiant/install/global.xlink"
error: xp_attach_gl_context returned: 2

(radiant.bin:5922): GdkGLExt-WARNING **: glXMakeCurrent() failed
[xcb] Unknown sequence number while processing queue
[xcb] Most likely this is a multi-threaded client and XInitThreads has not been called
[xcb] Aborting, sorry about that.
Assertion failed: (! xcb_xlib_threads_sequence_lost), function poll_for_event, file xcb_io.c, line 274.
Abort trap: 6

The workaround is to use the default layout, which places the texture window in the bottom-right corner of the main window.

**Comment by CircaCitadel** [comment]
Will this ever be fixed?

**Comment by jdolan** [comment]
I think there are some GL context sharing options that need to be enabled for this to work safely. I briefly poked at it a couple years ago, but.. short answer: I have no plans to fix this. I've learned to live with the default layout.

**Comment by CircaCitadel** [comment]
Fair enough. I found that you can press N or O to bring up the Entity or Console window, and switch to the Texture window via the tabs at the bottom. That's what I've been doing.

**Comment by jdolan** [comment]
I can't recall if NetRadiant has the same issue on OS X.. Might be worth looking at, both as a workaround for users who strongly prefer the floating texture window, and for possibly porting a fix over to GtkRadiant.

I have a version of NetRadiant that compiles and runs on OS X here:
https://github.com/jdolan/NetRadiant
